src/groups/mqb/mqbblp/mqbblp_clusterqueuehelper.cpp

```diff
@@ -3584,7 +3596,7 @@ void ClusterQueueHelper::restoreStateRemote()
         if (!isQueueAssigned(*queueContext.get())) {
             // Queue is not assigned to a partition; get it assigned.
 
-            assignQueue(queueContext);
+            assignQueueIfNeeded(cit.status(), queueContext);
```

Comment:
Note that there is a line 3596: `if (!isQueueAssigned(*queueContext.get())) {`
There is a second check in `assignQueueIfNeeded`: `if (!isQueueAssigned(*queueContext_sp)) { `
There is a third check in `assignQueue`: `BSLS_ASSERT_SAFE(!isQueueAssigned(*queueContext));`

All these checks might be checked sequentially when it's not really necessary.